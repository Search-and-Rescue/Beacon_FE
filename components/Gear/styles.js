const styles = {
  gearContainer: {
    alignItems: "center",
    flex: 1,
    marginHorizontal: "10%",
    marginVertical: 20
  },
  label: {
    color: "#001028",
    fontFamily: "Georgia",
    fontSize: 18,
    paddingBottom: 2,
    textAlign: "left",
    width: "100%"
  },
  input: {
    backgroundColor: "#F0F0F0",
    borderColor: "#001028",
    borderRadius: 2,
    borderWidth: 2,
    marginBottom: 15,
    padding: 10,
    width: "100%"
  },
  updateBtn: {
    backgroundColor: "#001028",
    borderColor: "#EFB095",
    borderWidth: 4,
    marginVertical: 10,
    paddingVertical: 12,
    width: "100%"
  },
  updateBtnText: {
    color: "#F0F0F0",
    fontFamily: "Futura",
    fontSize: 21,
    paddingHorizontal: '21%',
    textAlign: "center",
    width: "100%"
  }
};

export default styles;