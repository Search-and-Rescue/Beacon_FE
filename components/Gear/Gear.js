import React, { Component } from "react";
import { View, Text, TextInput } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { addGearItem, getGear } from '../../util/apiCalls';
import { connect } from "react-redux";
import styles from "./styles";
import { setGear } from '../../actions';
import { bindActionCreators } from 'redux';

export class Gear extends Component {
  constructor(props) {
    super(props);
    if (this.props.navigation.state.params === undefined) {
      this.props.navigation.state.params = { gear: {} }
    }
    const { gear } = this.props.navigation.state.params;
    this.state = {
      itemName: gear.itemName || "",
      description: gear.description || ""
    };
  }

  handleSubmit = async () => {
    const { navigation, user } = this.props;
    const newGearItem = {
      ...this.state,
      userId: parseInt(user.id)
    }
    await addGearItem(newGearItem);
    const userInfoGear = await getGear();
    this.props.setGear(userInfoGear.user.gear);
    navigation.navigate("GearList");
    this.setState({
      itemName: "",
      description: ""
    });
  };

  render() {
    return (
      <View style={styles.gearContainer}>
        <Text style={styles.label}>Item name:</Text>
        <TextInput
          placeholder="Red down sleeping bag"
          style={styles.input}
          onChangeText={text => this.setState({ itemName: text })}
          value={this.state.itemName}
          name="itemName"
        />
        <Text style={styles.label}>Item description:</Text>
        <TextInput
          placeholder="Mummy bag, 0 degree"
          style={styles.input}
          onChangeText={text => this.setState({ description: text })}
          value={this.state.description}
          name="description"
        />
        <TouchableOpacity style={styles.updateBtn} onPress={this.handleSubmit}>
          <Text style={styles.updateBtnText}>ADD</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export const mapStateToProps = ({ user }) => ({
  user
});

export const mapDispatchToProps = dispatch => bindActionCreators({ setGear }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Gear);