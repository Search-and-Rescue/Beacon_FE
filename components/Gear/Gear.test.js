import "react-native";
import React from "react";
import { shallow } from "enzyme";
import Gear from "./Gear";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

jest.mock('react-native-gesture-handler', () => "TouchableOpacity");

describe("Gear", () => {

  let wrapper, mockGear;
  beforeEach(() => {
    mockProps = {
      navigation: {
        state: {
          params: {
            gear : {
              id: 1,
              gear_name: "tent",
              description: "REI red 2 person"
            }
          }
        }
      }
    }
  });

  it("should match the snapshot with all of the data passed through", () => {
    expect(shallow(<Gear {...mockProps}/>)).toMatchSnapshot();
  });
});
