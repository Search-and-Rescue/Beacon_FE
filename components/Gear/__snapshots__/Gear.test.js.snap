// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gear should match the snapshot with all of the data passed through 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginHorizontal": "10%",
      "marginVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#001028",
        "fontFamily": "Georgia",
        "fontSize": 18,
        "paddingBottom": 2,
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    Item name:
  </Text>
  <TextInput
    allowFontScaling={true}
    name="itemName"
    onChangeText={[Function]}
    placeholder="Red down sleeping bag"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#F0F0F0",
        "borderColor": "#001028",
        "borderRadius": 2,
        "borderWidth": 2,
        "marginBottom": 15,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "color": "#001028",
        "fontFamily": "Georgia",
        "fontSize": 18,
        "paddingBottom": 2,
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    Item description:
  </Text>
  <TextInput
    allowFontScaling={true}
    name="description"
    onChangeText={[Function]}
    placeholder="Mummy bag, 0 degree"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#F0F0F0",
        "borderColor": "#001028",
        "borderRadius": 2,
        "borderWidth": 2,
        "marginBottom": 15,
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value="REI red 2 person"
  />
  <Component
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#001028",
        "borderColor": "#EFB095",
        "borderWidth": 4,
        "marginVertical": 10,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#F0F0F0",
          "fontFamily": "Futura",
          "fontSize": 21,
          "paddingHorizontal": "24%",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      ADD GEAR ITEM
    </Text>
  </Component>
</View>
`;
